test/test.go

```diff
@@ -0,0 +1,213 @@
+package test
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/bitrise-io/bitrise/models"
+	"github.com/bitrise-io/go-utils/fileutil"
+	"github.com/bitrise-io/steps-deploy-to-bitrise-io/test/converters"
+	"github.com/google/uuid"
+)
+
+func findImages(testDir string) (imageFilePaths []string) {
+	for _, ext := range []string{".jpg", ".jpeg", ".png"} {
+		if paths, err := filepath.Glob(filepath.Join(testDir, "*"+ext)); err == nil {
+			imageFilePaths = append(imageFilePaths, paths...)
+		}
+		if paths, err := filepath.Glob(filepath.Join(testDir, "*"+strings.ToUpper(ext))); err == nil {
+			imageFilePaths = append(imageFilePaths, paths...)
+		}
+	}
+	return
+}
+
+// ResultAssets ...
+type ResultAssets struct {
+	XMLContent []byte                    `json:"xml"`
+	ImagePaths []string                  `json:"-"`
+	StepInfo   models.TestResultStepInfo `json:"step_info"`
+}
+
+// Bundle ...
+type Bundle []ResultAssets
+
+// ParseTestResults ...
```

Comment:
Lists dirs from the rootTestsDir, finds all files from the next level then iterates through all the converters and to this dirs and appends the xml's and assets into a bundle if a converter is detected for a directory